File: java/PiedraPapelTijera.java
import java.util.HashMap;
import java.util.Scanner;

public class PiedraPapelTijera {

    public static void main(String[] args) {

        Scanner datos = new Scanner(System.in);

        HashMap<String, String> jugadasGanadoras = new HashMap<>();
        jugadasGanadoras.put("piedra", "tijera");
        jugadasGanadoras.put("papel", "piedra");
        jugadasGanadoras.put("tijera", "papel");

        boolean ganador = false;

        while (!ganador) {

            System.out.print("[Jugador 1] Escriba su opción (piedra, papel o tijera): ");
            String jugador1 = datos.nextLine().toLowerCase();

            System.out.print("[Jugador 2] Escriba su opción (piedra, papel o tijera): ");
            String jugador2 = datos.nextLine().toLowerCase();

            if (!jugadasGanadoras.containsKey(jugador1) || 
                !jugadasGanadoras.containsKey(jugador2)) {

                System.out.println("Alguna entrada es inválida, vuelve a intentarlo");
                continue;
            }

            if (jugador1.equals(jugador2)) {
                System.out.println("¡Empate!");
                ganador = false;

            } else if (jugadasGanadoras.get(jugador1).equals(jugador2)) {
                System.out.println("¡Jugador 1 gana!");
                ganador = true;

            } else {
                System.out.println("¡Jugador 2 gana!");
                ganador = true;
            }
        }

        datos.close();
    }
}

File: python/PiedraPapelTijera.py
gana = False

# reglas del juego: qué le gana a qué
ganadas = {
    "piedra": "tijera",
    "tijera": "papel",
    "papel": "piedra"
}

while not gana:
    print("Turno del jugador 1")
    print("piedra | papel | tijera")
    jugador1 = input().lower()

    print("Turno del jugador 2")
    print("piedra | papel | tijera")
    jugador2 = input().lower()

    # validar entradas
    if jugador1 not in ganadas or jugador2 not in ganadas:
        print("Entrada inválida, intenten otra vez\n")
        continue

    # empate
    if jugador1 == jugador2:
        print("Empate, vuelvan a jugar\n")

    # jugador 1 gana
    elif ganadas[jugador1] == jugador2:
        print("Gana jugador 1")
        gana = True

    # jugador 2 gana
    else:
        print("Gana jugador 2")
        gana = True